### PR TITLE
Target Postgres additions

### PR DESCRIPTION
**Pull request by BuzzCutNorman**
Here is some code I have been using in a Postgres target I use for internal development.  Please take a look and let me know if any of these additions would be helpful.

### CODE REVIEW

File: target_postgres/connector.py [hunk 1]
```diff
@@ -24,3 +26,29 @@ def create_sqlalchemy_connection(self) -> sqlalchemy.engine.Connection:
...
         """
         return self.create_sqlalchemy_engine().connect()
+    
+    def create_sqlalchemy_engine(self) -> sqlalchemy.engine.Engine:
+        """Return a new SQLAlchemy engine using the provided config.
+        
+        Returns:
+            A newly created SQLAlchemy engine object.
+        """
+        return sqlalchemy.create_engine(
```
Comment (visch) [inline target_postgres/connector.py:36]: Nice I'm not sure about `values_plus_batch` is there a reason you added that here? 

future=True is interesting as well, might be best to just jump to the new versions!

**Comment by visch** [comment]
I'm going to keep this code in mind for https://github.com/MeltanoLabs/target-postgres/pull/17 I'll merge it into that branch or merge this later. Generally this looks good I just want to get a good test run going first and https://github.com/MeltanoLabs/target-postgres/pull/17 is adding a good amount of tests. Merging the two together may be the best option! Still looking

File: target_postgres/connector.py (hunk 1 above)
Comment (BuzzCutNorman) [reply target_postgres/connector.py:36]: I put the `values_plus_batch` there to match the examples SQLAlchemy gave for postgres fast execution helpers.  https://docs.sqlalchemy.org/en/14/dialects/postgresql.html#psycopg2-fast-execution-helpers

The `future=True` is to assist in migration from 1.4 to 2.0.  Here is the link I got that from. https://docs.sqlalchemy.org/en/14/changelog/migration_20.html#migration-to-2-0-step-four-use-the-future-flag-on-engine

File: target_postgres/connector.py [hunk 2]
```diff
@@ -24,3 +26,29 @@ def create_sqlalchemy_connection(self) -> sqlalchemy.engine.Connection:
...
+
+    @staticmethod
+    def to_sql_type(jsonschema_type: dict) -> sqlalchemy.types.TypeEngine:
+        """Returns a JSON Schema equivalent for the given SQL type.
+        
+        Developers may optionally add custom logic before calling the default implementation
+        inherited from the base class.
+        """
+        # Convert date-time to timestamp
+        if jsonschema_type.get('format') == 'date-time':
```
Comment (visch) [inline target_postgres/connector.py:51]: This is nice! Took it!

File: target_postgres/sinks.py [hunk 3]
```diff
@@ -8,3 +13,62 @@ class PostgresSink(SQLSink):
...
+        """Generate an insert statement for the given records.
+
+        Args:
+            full_table_name: the target table name.
+            schema: the JSON schema for the new table.
+
+        Returns:
+            An insert statement.
+        """
+        statement = insert(self.connector.get_table(full_table_name))
```
Comment (visch) [inline target_postgres/sinks.py:31]: +1 this is integrated in more or less!

File: target_postgres/sinks.py [hunk 4]
```diff
@@ -8,3 +13,62 @@ class PostgresSink(SQLSink):
...
+            True if table exists, False if not, None if unsure or undetectable.
+        """
+        insert_sql = self.generate_insert_statement(
+            full_table_name,
+            schema,
+        )
+
+        try:
+            with self.connector.connection.engine.begin() as conn:
+                conn.execute(
```
Comment (visch) [inline target_postgres/sinks.py:63]: +1 this is generally worked in!

**Comment by BuzzCutNorman** [comment]
@visch sounds like there were a few things that were helpful.  Did you want me to close this PR or would you like me to pair it down to only items you want?

**Comment by visch** [comment]
@BuzzCutNorman  Thank you very much for putting this in!

**Comment by BuzzCutNorman** [comment]
@visch your welcome I am glad it had some usefully items for you

**Pull request closed without merging**